### ISSUE TITLE
Progress indicator: Design system website contextual example

### ISSUE THREAD

**Issue by Spark450** [GovAlta/ui-components#2123]
## Acceptance Criteria:

- Add jump links:
  - Show progress when loading a page
  - Show progress when loading a table with lots of data
  - Show progress while uploading a file
- Add the following examples and code snippets ([Figma design](https://www.figma.com/design/aeIJGExcpx9ecqnIVwImi1/Component-Pages?node-id=88-371479&t=wm8tst8hY9zJU4Cd-1))
  - Show progress when loading a page (Clarify design intent with @twjeffery before implementation)
  - Show progress when loading a table with lots of data
  - Show progress while uploading a file

**Comment by ArakTaiRoth** [comment]
I don't believe these examples are necessary, and indeed there's never been any issues surfaced with the progress indicator. All these examples are the progress indicator in different containers, that's easy enough to extrapolate with our current examples on the main page.
